pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -55,7 +55,7 @@ in
 , changelog ? null
 , mainProgram ? null
 , doCoverage ? false
-, doHaddock ? !(ghc.isHaLVM or false)
+, doHaddock ? !(ghc.isHaLVM or false) && ghc.hasHaddock
```

Comment:
The remaining GHCs (8.8.4, 8.10.7, HEAD) need this annotation as well, e. g. `ghc*Binary`, but also `ghcjs`, see the current eval error as per ofborg.

Reply:
The binary compilers don't have the `enableHaddockProgram` arguments. What should I set `hasHaddock` to?

Reply:
What about just doing `&& (ghc.hasHaddock or true)`?

Reply:
> What about just doing `&& (ghc.hasHaddock or true)`?

Also works and probably a smart move since it means that we don't have to commit to having this attribute forever.

Reply:
Ok, did it this way. I also added `hasHaddock` to `ghcHEAD`, so the main versions are covered.

Reply:
The evaluation is fixed, now.

Reply:
GHC 8.8.4 and 8.10.7 also have an `enableHaddockProgram` argument, don't they?

Reply:
Yes, but I though it would be enough to target the next major version 9.0 and head.
Do you want me to add those too?

Reply:
Yeah, definitely, the package sets are still around and work for the most part and they should behave like the default ones as much as possible!

Reply:
Done.